client/components/create-profile/LinksSection.tsx

```diff
@@ -11,191 +11,113 @@ import {
-                    <FontAwesomeIcon icon={faLinkedinIn} />
-                  </TagLabel>
-                </Tag>
-              </Link>
-            </NextLink>
-          ) : null}
+export const ProfileLink = (props: ProfileLinkProps) => (
+  <NextLink href={props.href} passHref>
+    <Link target="_blank">
+      <Tag
+        w="fit-content"
+        size="lg"
+        borderRadius="8px"
+        bgColor="#E2E9F0"
+        color="black"
+      >
+        <TagLabel m="auto">
+          <FontAwesomeIcon icon={props.icon} />
+        </TagLabel>
+      </Tag>
+    </Link>
+  </NextLink>
+);
 
-          {producthunt ? (
-            <NextLink
-              href={`https://www.producthunt.com/@${producthunt}`}
-              passHref
-            >
-              <Link target="_blank">
-                <Tag
-                  cursor="pointer"
-                  w="fit-content"
-                  size="lg"
-                  borderRadius="8px"
-                  bgColor="#E2E9F0"
-                  color="black"
-                >
-                  <TagLabel m="auto">
-                    <FontAwesomeIcon icon={faProductHunt} />
-                  </TagLabel>
-                </Tag>
-              </Link>
-            </NextLink>
-          ) : null}
+type LinksSectionProps = {
+  profileLinks?: {
+    linkedin?: string;
+    twitter?: string;
+    behance?: string;
+    dribbble?: string;
+    producthunt?: string;
+    github?: string;
+    dev?: string;
+    medium?: string;
+  };
+};
 
-          {dribbble ? (
-            <NextLink href={`https://www.dribbble.com/${dribbble}`} passHref>
-              <Link target="_blank">
-                <Tag
-                  cursor="pointer"
-                  w="fit-content"
-                  size="lg"
-                  borderRadius="8px"
-                  bgColor="#E2E9F0"
-                  color="black"
-                >
-                  <TagLabel m="auto">
-                    <FontAwesomeIcon icon={faDribbble} />
-                  </TagLabel>
-                </Tag>
-              </Link>
-            </NextLink>
-          ) : null}
+export default function LinksSection(props: LinksSectionProps) {
+  if (!props.profileLinks) return null;
 
-          {behance ? (
-            <NextLink href={`https://www.behance.net/${behance}`} passHref>
-              <Link target="_blank">
-                <Tag
-                  cursor="pointer"
-                  w="fit-content"
-                  size="lg"
-                  borderRadius="8px"
-                  bgColor="#E2E9F0"
-                  color="black"
-                >
-                  <TagLabel m="auto">
-                    <FontAwesomeIcon icon={faBehance} />
-                  </TagLabel>
-                </Tag>
-              </Link>
-            </NextLink>
-          ) : null}
+  const links = Object.keys(props.profileLinks).map<ProfileLinkProps | null>(
+    (key) => {
+      // TS thinks that we can get here somehow...
```

Comment:
TS thinks `props.profileLinks` can be undefined allthough we return in the line above hence the `!`

Reply:
I would suggest is to use null coalescing for this one rather than it being `null`

Thus

```ts
Object.keys(props.profileLinks || {})
```

Reply:
I mean, opposed to using JS-logic (and thus adding bundled code) for a TS-error but I'm fine if that warrants a "request change"

Reply:
Oh lol think I have it misunderstood, I was referring to this since it is what you have your comment on
```ts
const links = Object.keys(props.profileLinks).map<ProfileLinkProps | null>(
```
But you were referring to something else

mb!

Reply:
It doesn't look pretty atm, can we revert it instead :)